pootle/apps/pootle_store/syncer.py

```diff
@@ -0,0 +1,365 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import logging
+import os
+from collections import namedtuple
+
+from translate.storage.factory import getclass
+
+from django.utils.functional import cached_property
+
+from pootle.core.delegate import format_classes
+from pootle.core.log import log
+
+from .constants import OBSOLETE
+from .models import Unit
+from .util import get_change_str
+
+
+class UnitSyncer(object):
+
+    def __init__(self, unit):
+        self.unit = unit
+
+    @property
+    def context(self):
+        return self.unit.getcontext()
+
+    @property
+    def isobsolete(self):
+        return self.unit.isobsolete()
+
+    @property
+    def isfuzzy(self):
+        return self.unit.isfuzzy()
+
+    @property
+    def locations(self):
+        return self.unit.getlocations()
+
+    @property
+    def developer_notes(self):
+        return self.unit.getnotes(origin="developer")
+
+    @property
+    def source(self):
+        return self.unit.source
+
+    @property
+    def target(self):
+        return self.unit.target
+
+    @property
+    def translator_notes(self):
+        return self.unit.getnotes(origin="translator")
+
+    @property
+    def unitid(self):
+        return self.unit.getid()
+
+    def convert(self, unitclass):
+        newunit = self.create_unit(unitclass)
+        self.set_target(newunit)
+        self.set_fuzzy(newunit)
+        self.set_locations(newunit)
+        self.set_notes(newunit)
+        self.set_unitid(newunit)
+        self.set_context(newunit)
+        self.set_obsolete(newunit)
+        return newunit
+
+    def create_unit(self, unitclass):
+        return unitclass(self.source)
+
+    def set_developer_notes(self, newunit):
+        notes = self.developer_notes
+        if notes:
+            newunit.addnote(notes, origin="developer")
+
+    def set_fuzzy(self, newunit):
+        newunit.markfuzzy(self.isfuzzy)
+
+    def set_unitid(self, newunit):
+        newunit.setid(self.unitid)
+
+    def set_context(self, newunit):
+        newunit.setcontext(self.context)
+
+    def set_locations(self, newunit):
+        locations = self.locations
+        if locations:
+            newunit.addlocations(locations)
+
+    def set_obsolete(self, newunit):
+        if self.isobsolete:
+            newunit.makeobsolete()
+
+    def set_notes(self, newunit):
```

Comment:
Instead of adding this method that just calls two other methods, just get rid of this method and replace any call to this method by the two calls to the other two methods.